static/app/utils/useDismissAlert.tsx

```diff
@@ -0,0 +1,60 @@
+import {useCallback} from 'react';
+
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+
+type Opts = {
+  /**
+   * Key in localstorage.
+   * Use a format like: `${organization.id}:my-feature`
+   */
+  key: string;
+
+  /**
+   * Number of days before the dismissal is expired.
+   * After expiration the the user will need to re-dismiss things.
+   */
+  expirationDays?: number;
+};
+
+const MS_PER_DAY = 1000 * 60 * 60 * 24;
+
+function isValid(timestamp: number, expirationDays: number) {
+  if (!timestamp || isNaN(timestamp)) {
+    return false;
+  }
+  const duration = Date.now() - timestamp;
+  const days = duration / MS_PER_DAY;
+  return days < expirationDays;
+}
+
+/**
+ * Nominally for tracking dismissal/acknowledgement of in-app alerts and
+ * notifications, specfically those that don't need to be persisted server side.
+ *
+ * Dismissal can be 'permanent' or have an expiration after some number of days.
+ *
+ * The real lifecycle depends on whether users reset localStorage or not, if you
+ * need something really permanent then save the users' preference to the server.
+ */
+function useDismissAlert({expirationDays = Number.MAX_SAFE_INTEGER, key}: Opts) {
+  const [dismissedTimestamp, setDismissedTimestamp] = useLocalStorageState(
+    key,
+    (_val, raw) => raw
+  );
+
+  const isDismissed =
+    expirationDays === Number.MAX_SAFE_INTEGER
+      ? Boolean(dismissedTimestamp)
+      : isValid(Number(dismissedTimestamp), expirationDays);
+
+  const dismiss = useCallback(() => {
+    setDismissedTimestamp(Date.now().toString());
+  }, [setDismissedTimestamp]);
+
+  return {
+    isDismissed,
+    dismiss,
+  };
+}
+
+export default useDismissAlert;
```

Comment:
nit: shall we make this `named export`?

Reply:
If that's not a hard rule i'm not really interested in doing it. I have opinions on named exports but this isn't the place for that

Reply:
agree...maybe that is something for the Frontend TSC